### PR TITLE
Rebranding docs/interact/

### PR DESCRIPTION
**Pull request by zuiderkwast**
Replaced Valkey with Redis and similar under this directory.

Commands used, one per commit:

```
git grep-replace -P 'redis(?=\.call|\.pcall|\.regis|\.set_repl)' server
git grep-replace -P 'Redis(?= compatibility| v?[2-7]| version| < [2-7])' 'Redis OSS'
git grep-replace -P 'Redis(?! compatibility| v?[2-7]| version| < [2-7]| OSS|\.[a-z])' Valkey
git grep-replace -F 'https://github.com/redis/redis/blob/unstable/' 'https://github.com/valkey-io/valkey/blob/unstable/'
git grep-replace  -P 'redis(?=\.replicate|\.setresp|\.log|\.error_reply|\.status_reply|\.sha1hex|\.breakpoint|\.debug|\.acl_check_cmd)' server
git grep-replace -F redis.conf valkey.conf
```

Not done: Remove references to old Redis versions. That can be done later as it requires more manual work. The regexes used here proved pretty solid for almost-automatic rebranding.

### CODE REVIEW

**Comment by zuiderkwast** [comment]
Part of #18. @SoulPancake FYI.

**Review by PingXie — APPROVED** [review]
I didn't validate the hyper links but otherwise LGTM.

Also, same comment/question on "OSS Redis" vs "Redis OSS". So we are settling on "Redis OSS"? I am fine with that but just wanted to make this more explicit and ensure there is a consensus among the team, cc @valkey-io/core-team.

**Comment by zuiderkwast** [comment]
Regarding "OSS Redis" vs "Redis OSS", do we really have a choice? The trademark policy is pretty clear, but you don't agree?

File: docs/interact/programmability/lua-api.md [hunk 1]
```diff
@@ -344,17 +344,17 @@ It accepts a single input argument that only be one of the following:
...
 A simple example follows:
 
 ```lua
-redis.replicate_commands() -- Enable effects replication in versions lower than Redis v7.0
-redis.call('SET', KEYS[1], ARGV[1])
-redis.set_repl(redis.REPL_NONE)
-redis.call('SET', KEYS[2], ARGV[2])
-redis.set_repl(redis.REPL_ALL)
-redis.call('SET', KEYS[3], ARGV[3])
+redis.replicate_commands() -- Enable effects replication in versions lower than Redis OSS v7.0
```
Comment (zuiderkwast) [inline docs/interact/programmability/lua-api.md:352]: Obviously I missed some Lua functions. I'll add another commit with those.

redis.replicate_commands
redis.debug
redis.acl_check_cmd

and some more

**Pull request merged**